connectivity/netsocket/include/netsocket/InternetDatagramSocket.h

```diff
@@ -74,6 +75,58 @@ class InternetDatagramSocket : public InternetSocket {
     nsapi_size_or_error_t recvfrom(SocketAddress *address,
                                    void *data, nsapi_size_t size) override;
 
+    /** Send datagram and ancillary data to the specified address.
+     *
+     *  By default, sendto blocks until data is sent. If socket is set to
+     *  nonblocking or times out, NSAPI_ERROR_WOULD_BLOCK is returned
+     *  immediately.
+     *
+     *  It uses sendmsg with zero ancillary data
+     *  @param address  The SocketAddress of the remote host.
+     *  @param data     Buffer of data to send to the host.
+     *  @param size     Size of the buffer in bytes.
+     *  @param control     Size of the buffer in bytes.
+     *  @param control_size     Size of the buffer in bytes.
+     *  @retval         NSAPI_ERROR_NO_SOCKET in case socket was not created correctly.
+     *  @retval         NSAPI_ERROR_WOULD_BLOCK in case non-blocking mode is enabled
+     *                  and send cannot be performed immediately.
+     *  @retval         int Other negative error codes for stack-related failures.
+     *                  See \ref NetworkStack::socket_send.
+     */
+    nsapi_size_or_error_t sendmsg(const SocketAddress &address,
+                                  const void *data, nsapi_size_t size,
+                                  nsapi_msghdr_t *control, nsapi_size_t control_size) override;
+
+
+    /** Receive a datagram with ancillary data and store the source address in address if it's not NULL.
+    *
+    *  By default, recvfrom blocks until a datagram is received. If socket is set to
+    *  nonblocking or times out with no datagram, NSAPI_ERROR_WOULD_BLOCK
+    *  is returned.
```

Comment:
Same as above, this is recvmsg and not recvfrom.

Reply:
Done